### ISSUE TITLE
remote ssh error

### ISSUE THREAD

**Issue by JavanTang** [microsoft/vscode-remote-release#6437]
There is no problem with connecting to the server directly using terminals, but there is a problem with using vs remote ssh

- VSCode Version: 1.65.2
- Local OS Version: macOS 12.3
- Remote OS Version: Ubuntu 20.04.3 LTS (GNU/Linux 5.13.0-30-generic x86_64)
- Remote Extension/Connection Type: SSH
- Logs:
```
[16:58:00.965] Log Level: 2
[16:58:00.966] remote-ssh@0.76.1
[16:58:00.966] darwin arm64
[16:58:00.979] SSH Resolver called for "ssh-remote+tong", attempt 1
[16:58:00.983] "remote.SSH.useLocalServer": true
[16:58:00.983] "remote.SSH.path": undefined
[16:58:00.983] "remote.SSH.configFile": undefined
[16:58:00.983] "remote.SSH.useFlock": true
[16:58:00.983] "remote.SSH.lockfilesInTmp": false
[16:58:00.983] "remote.SSH.localServerDownload": auto
[16:58:00.983] "remote.SSH.remoteServerListenOnSocket": false
[16:58:00.983] "remote.SSH.showLoginTerminal": false
[16:58:00.984] "remote.SSH.defaultExtensions": []
[16:58:00.984] "remote.SSH.loglevel": 2
[16:58:00.984] "remote.SSH.enableDynamicForwarding": true
[16:58:00.984] "remote.SSH.enableRemoteCommand": false
[16:58:00.984] "remote.SSH.serverPickPortsFromRange": {}
[16:58:00.984] "remote.SSH.serverInstallPath": {}
[16:58:00.992] SSH Resolver called for host: tong
[16:58:00.992] Setting up SSH remote "tong"
[16:58:00.995] Acquiring local install lock: /var/folders/cr/mbb464xs5c3_bpg49x4bk5kc0000gn/T/vscode-remote-ssh-bf1a3424-install.lock
[16:58:01.001] Looking for existing server data file at /Users/tangzhifeng/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-bf1a3424-c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1-0.76.1/data.json
[16:58:01.003] Using commit id "c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1" and quality "stable" for server
[16:58:01.005] Install and start server if needed
[16:58:01.010] PATH: /Library/TeX/texbin/:/opt/homebrew/opt/openjdk/bin:/Users/tangzhifeng/.poetry/bin:/Users/tangzhifeng/miniforge3/bin:/Users/tangzhifeng/miniforge3/condabin:/Library/Frameworks/Python.framework/Versions/3.9/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Library/Apple/usr/bin:/Users/tangzhifeng/.poetry/bin:/Users/tangzhifeng/Tools/flutter/bin
[16:58:01.010] Checking ssh with "ssh -V"
[16:58:01.022] > OpenSSH_8.6p1, LibreSSL 3.3.5

[16:58:01.036] askpass server listening on /var/folders/cr/mbb464xs5c3_bpg49x4bk5kc0000gn/T/vscode-ssh-askpass-f10b26bb904811451d14c55d46df3132941f16ee.sock
[16:58:01.036] Spawning local server with {"serverId":1,"ipcHandlePath":"/var/folders/cr/mbb464xs5c3_bpg49x4bk5kc0000gn/T/vscode-ssh-askpass-abf7335ddf5311e467863278358204588c564d64.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","52562","-o","ConnectTimeout=15","tong"],"serverDataFolderName":".vscode-server","dataFilePath":"/Users/tangzhifeng/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-bf1a3424-c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1-0.76.1/data.json"}
[16:58:01.036] Local server env: {"SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.DG2nctJsXx/Listeners","SHELL":"/bin/zsh","DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/MacOS/Electron","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/cr/mbb464xs5c3_bpg49x4bk5kc0000gn/T/vscode-ssh-askpass-f10b26bb904811451d14c55d46df3132941f16ee.sock"}
[16:58:01.039] Spawned 52426
[16:58:01.128] > local-server-1> Spawned ssh, pid=52436
[16:58:01.132] stderr> OpenSSH_8.6p1, LibreSSL 3.3.5
[16:58:01.489] stderr> debug1: Server host key: ssh-ed25519 SHA256:TS123aZ7f1241415hgMHJYBIftp6y7W5g
[16:58:02.111] > Welcome to Ubuntu 20.04.3 LTS (GNU/Linux 5.13.0-30-generic x86_64)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> 
> 87 updates can be applied immediately.
> To see these additional updates run: apt list --upgradable
> 
> Your Hardware Enablement Stack (HWE) is supported until April 2025.
> *** System restart required ***
[16:58:02.191] > ready: 134bcf1127ec
[16:58:02.262] > Linux 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022
[16:58:02.263] Platform: linux
[16:58:02.332] > /bin/bash
[16:58:02.332] Shell: bash
[16:58:02.461] > 134bcf1127ec: running
[16:58:02.461] stderr> cat: '/lib/x86_64-linux-gnu/libstdc++.so.6'$'\n''/lib/i386-linux-gnu/libstdc++.so.6': No such file or directory
[16:58:02.470] stderr> mainScriptWrapper:185: bad math expression: operand expected at `*100 +  '
[16:58:02.471] stderr> Transferred: sent 18744, received 4024 bytes, in 0.6 seconds
[16:58:02.471] stderr> Bytes per second: sent 30219.0, received 6487.5
[16:58:02.471] > Missing GLIBCXX >= 3.4.18!
> Found versions 
> local-server-1> ssh child died, shutting down
[16:58:02.473] Local server exit: 0
[16:58:02.473] Received install output: local-server-1> Spawned ssh, pid=52436
OpenSSH_8.6p1, LibreSSL 3.3.5
Welcome to Ubuntu 20.04.3 LTS (GNU/Linux 5.13.0-30-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

87 updates can be applied immediately.
To see these additional updates run: apt list --upgradable

Your Hardware Enablement Stack (HWE) is supported until April 2025.
*** System restart required ***
ready: 134bcf1127ec
Linux 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022
/bin/bash
134bcf1127ec: running
cat: '/lib/x86_64-linux-gnu/libstdc++.so.6'$'\n''/lib/i386-linux-gnu/libstdc++.so.6': No such file or directory
mainScriptWrapper:185: bad math expression: operand expected at `*100 +  '
Transferred: sent 18744, received 4024 bytes, in 0.6 seconds
Bytes per second: sent 30219.0, received 6487.5
Missing GLIBCXX >= 3.4.18!
Found versions 
local-server-1> ssh child died, shutting down

[16:58:02.473] Failed to parse remote port from server output
[16:58:02.474] Resolver error: Error: 
	at Function.Create (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:586041)
	at Object.t.handleInstallOutput (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:584693)
	at Object.e [as tryInstallWithLocalServer] (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:625033)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async /Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:644084
	at async Object.t.withShowDetailsEvent (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:647831)
	at async /Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:623505
	at async b (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:620011)
	at async Object.t.resolveWithLocalServer (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:623120)
	at async Object.t.resolve (/Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:645441)
	at async /Users/tangzhifeng/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:722872
[16:58:02.476] 
```

Steps to Reproduce:
1. remote-ssh: connect to host...
2. Select the server

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by tanhakabir** [comment]
Could you try using the pre-release version of the extension to see if it works for you? 

This seems like a duplicate of https://github.com/microsoft/vscode-remote-release/issues/6407

**Comment by JavanTang** [comment]
The pre-release version didn't fix my problem either, unfortunately I did it by lowering the demo ssh version and vscode version.

My current vscode version is: 1.64.2 and the remotessh version is: 0.70.0

**Comment by tanhakabir** [comment]
Hmm all the other issues I'm seeing similar to yours have been fixed by pre-release version `0.77.X`. Could you share with me your logs when you run this version?

Also is your remote shell configured with anything special?

**Comment by JavanTang** [comment]
> Hmm all the other issues I'm seeing similar to yours have been fixed by pre-release version `0.77.X`. Could you share with me your logs when you run this version?
> 
> Also is your remote shell configured with anything special?

Isn't the log posted above?

**Comment by tanhakabir** [comment]
The logs above isn't with pre-release version 0.77.X. It's 0.76.1.

**Comment by JavanTang** [comment]
I just tried it, the pre-release version works fine

```
[15:32:45.025] Log Level: 2
[15:32:45.026] remote-ssh@0.77.2022031115
[15:32:45.026] darwin arm64
[15:32:45.046] SSH Resolver called for "ssh-remote+XXXXX", attempt 1
[15:32:45.047] "remote.SSH.useLocalServer": true
[15:32:45.047] "remote.SSH.path": undefined
[15:32:45.047] "remote.SSH.configFile": undefined
[15:32:45.047] "remote.SSH.useFlock": true
[15:32:45.047] "remote.SSH.lockfilesInTmp": false
[15:32:45.048] "remote.SSH.localServerDownload": auto
[15:32:45.048] "remote.SSH.remoteServerListenOnSocket": false
[15:32:45.048] "remote.SSH.showLoginTerminal": false
[15:32:45.048] "remote.SSH.defaultExtensions": []
[15:32:45.048] "remote.SSH.loglevel": 2
[15:32:45.048] "remote.SSH.enableDynamicForwarding": true
[15:32:45.048] "remote.SSH.enableRemoteCommand": false
[15:32:45.049] "remote.SSH.serverPickPortsFromRange": {}
[15:32:45.049] "remote.SSH.serverInstallPath": {}
[15:32:45.056] SSH Resolver called for host: XXXXX
[15:32:45.056] Setting up SSH remote "XXXXXXX"
[15:32:45.059] Acquiring local install lock: XXXXXXX
[15:32:45.062] Looking for existing server data file at /Users/XXXXXXX/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/XXXXXXXX/data.json
[15:32:45.062] Using commit id "XDXD" and quality "stable" for server
[15:32:45.065] Install and start server if needed
[15:32:45.074] PATH: /Library/TeX/texbin/:/opt/homebrew/opt/openjdk/bin:/Users/XXXX/.poetry/bin:/Users/XXX/miniforge3/bin:/Users/CXCXCXCXC/miniforge3/condabin:/Library/Frameworks/Python.framework/Versions/3.9/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Library/Apple/usr/bin:/Users/CXCXCXCXC/.cargo/bin:/Users/CXCXCXCXC/.poetry/bin:/Users/CXCXCXCXC/Tools/flutter/bin:/Users/CXCXCXCXC/Tools/chrome
[15:32:45.075] Checking ssh with "ssh -V"
[15:32:45.086] > OpenSSH_8.6p1, LibreSSL 3.3.5

[15:32:45.089] askpass server listening on /var/folders/cr/XDXD/T/vscode-ssh-askpass-X.sock
[15:32:45.090] Spawning local server with {"serverId":1,"ipcHandlePath":"/var/folders/cr/XDXD/T/vscode-ssh-askpass-bc554dc209a2dc68a3090adef643f0ce65652b61.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","60484","-o","ConnectTimeout=15","XXXXXX"],"serverDataFolderName":".vscode-server","dataFilePath":"/Users/XXXXXXXX/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-X-X-0.77.2022031115/data.json"}
[15:32:45.090] Local server env: {"SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.flUCqbOOtw/Listeners","SHELL":"/bin/zsh","DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/CXCXCXCXC/.vscode/extensions/ms-vscode-remote.remote-ssh-0.77.2022031115/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/MacOS/Electron","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"/Users/CXCXCXCXC/.vscode/extensions/ms-vscode-remote.remote-ssh-0.77.2022031115/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/cr/XXXXXX/T/vscode-ssh-askpass-XXXXXX.sock"}
[15:32:45.091] Spawned 1237
[15:32:45.179] > local-server-1> Spawned ssh, pid=1247
[15:32:45.182] stderr> OpenSSH_8.6p1, LibreSSL 3.3.5
[15:32:45.863] stderr> Authenticated to XXXXXXXX
[15:32:46.441] > Welcome to Ubuntu 20.04.3 LTS (GNU/Linux 5.13.0-30-generic x86_64)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> 
> 100 updates can be applied immediately.
> To see these additional updates run: apt list --upgradable
> 
> Your Hardware Enablement Stack (HWE) is supported until April 2025.
> *** System restart required ***
[15:32:46.558] > ready: 026db93338c0
[15:32:46.634] > Linux 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022
[15:32:46.634] Platform: linux
[15:32:46.705] > /bin/bash
[15:32:46.705] Shell: bash
[15:32:46.705] Creating bash subshell inside "bash"
[15:32:46.778] > bash version:  5.0.17(1)-release
[15:32:46.831] > bash version:  5.0.17(1)-release
[15:32:46.888] > 026db93338c0: running
[15:32:46.938] > Acquiring lock on /home/XXXXXX/.vscode-server/bin/XDXD/vscode-remote-lock.XDXD.XDXD
[15:32:46.973] > Found existing installation at /home/XDXD/.vscode-server/bin/XDXD...
> Checking /home/XDXD/.vscode-server/.XDXD.log and /home/XDXD/.vscode-server/.XDXD.pid for a running server
[15:32:47.005] > Looking for server with pid: 2136706
[15:32:47.046] > Starting server with command... /home/XDXD/.vscode-server/bin/XDXD/bin/code-server --start-server --host=127.0.0.1 --accept-server-license-terms --enable-remote-auto-shutdown --port=0 --telemetry-level all   &> "/home/XDXD/.vscode-server/.XDXD.log" < /dev/null
> printenv:
[15:32:47.047] >     SHELL=/bin/bash
>     LC_ADDRESS=zh_CN.UTF-8
>     LC_NAME=zh_CN.UTF-8
>     LC_MONETARY=zh_CN.UTF-8
>     PWD=/home/XXXXXX
>     LOGNAME=XXXXXXXX
>     XDG_SESSION_TYPE=tty
>     MOTD_SHOWN=pam
>     HOME=/home/XXXXX
>     LANG=en_US.UTF-8
>     LC_PAPER=zh_CN.UTF-8
>     VSCODE_AGENT_FOLDER=/home/XXXX/.vscode-server
>     SSH_CONNECTION=127.0.0.1 42582 127.0.0.1 22
>     XDG_SESSION_CLASS=user
>     LC_IDENTIFICATION=zh_CN.UTF-8
>     USER=XDXD
>     SHLVL=1
>     LC_TELEPHONE=zh_CN.UTF-8
>     LC_MEASUREMENT=zh_CN.UTF-8
>     XDG_SESSION_ID=2684
>     XDG_RUNTIME_DIR=/run/user/1010
>     HOMEBREW_BREW_GIT_REMOTE=https://mirrors.ustc.edu.cn/brew.git
>     SSH_CLIENT=127.0.0.1 42582 22
>     LC_TIME=zh_CN.UTF-8
>     XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
>     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
>     DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1010/bus
>     LC_NUMERIC=zh_CN.UTF-8
>     OLDPWD=/home/XXXXXXX
>     _=/usr/bin/printenv
[15:32:47.057] > Removing old logfile at /home/XDXD/.vscode-server/.XDXD.log
[15:32:47.072] > Spawned remote server: 2872307
[15:32:47.099] > Waiting for server log...
[15:32:47.122] > Waiting for server log...
[15:32:47.145] > Waiting for server log...
[15:32:47.180] > Waiting for server log...
[15:32:47.234] > Waiting for server log...
[15:32:47.253] > Waiting for server log...
[15:32:47.287] > Waiting for server log...
[15:32:47.329] > Waiting for server log...
[15:32:47.360] > Waiting for server log...
[15:32:47.397] > Waiting for server log...
[15:32:47.433] > Waiting for server log...
[15:32:47.469] > Waiting for server log...
[15:32:47.505] > Waiting for server log...
[15:32:47.541] > Waiting for server log...
[15:32:47.577] > Waiting for server log...
[15:32:47.619] > Waiting for server log...
[15:32:47.649] > Waiting for server log...
[15:32:47.685] > Waiting for server log...
[15:32:47.721] > Waiting for server log...
[15:32:47.758] > Waiting for server log...
[15:32:47.793] > Waiting for server log...
[15:32:47.828] > Waiting for server log...
[15:32:47.866] > Waiting for server log...
[15:32:47.905] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
[15:32:47.913] > 026db93338c0: start
> SSH_AUTH_SOCK====
[15:32:47.916] > DISPLAY====
> webUiAccessToken====
> listeningOn==35781==
> osReleaseId==ubuntu==
> arch==x86_64==
> tmpDir==/run/user/1010==
> platform==linux==
[15:32:47.921] > unpackResult====
> didLocalDownload==0==
> downloadTime====
[15:32:47.925] > installTime====
> extInstallTime====
> serverStartTime==854==
> connectionToken==a1111111-a111-111a-a111-111a1aa11a1a==
> 026db93338c0: end
[15:32:47.926] Received install output: 
SSH_AUTH_SOCK====DISPLAY====
webUiAccessToken====
listeningOn==35781==
osReleaseId==ubuntu==
arch==x86_64==
tmpDir==/run/user/1010==
platform==linux==unpackResult====
didLocalDownload==0==
downloadTime====installTime====
extInstallTime====
serverStartTime==854==
connectionToken==a1111111-a111-111a-a111-111a1aa11a1a==

[15:32:47.926] Remote server is listening on 35781
[15:32:47.926] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":35781},"osReleaseId":"ubuntu","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/1010","platform":"linux","connectionToken":"CCZCASF"},"serverStartTime":854,"installUnpackCode":""}
[15:32:47.927] Persisting server connection details to /Users/CXCXCXCXC/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-c219dd74-XDXD-0.77.2022031115/data.json
[15:32:47.929] Starting forwarding server. localPort 60498 -> socksPort 60484 -> remotePort 35781
[15:32:47.929] Forwarding server listening on 60498
[15:32:47.929] Waiting for ssh tunnel to be ready
[15:32:47.931] [Forwarding server 60498] Got connection 0
[15:32:47.931] Tunneled 35781 to local port 60498
[15:32:47.932] Resolved "ssh-remote+XXXXXX" to "127.0.0.1:60498"
[15:32:47.937] ------




[15:32:47.962] [Forwarding server 60498] Got connection 1
[15:32:48.734] [Forwarding server 60498] Got connection 2
[15:32:52.249] [Forwarding server 60498] Got connection 3
```
